### PR TITLE
add custom server params

### PR DESCRIPTION
**Pull request by esmeetu**
This PR add support to openai server with custom params: `min_p` and `repetition_penalty`. Although those are not official params, but i think it's useful to add those as extended params.

### CODE REVIEW

File: vllm/sampling_params.py [hunk 1]
```diff
@@ -149,6 +149,7 @@ def __init__(
                 # Zero temperature means greedy sampling.
                 self.top_p = 1.0
                 self.top_k = -1
+                self.min_p = 0.0
```
Comment (esmeetu) [inline vllm/sampling_params.py:152]: same as top_p&top_k, min_p is unneccessary when using greedy sampling.

**Comment by WoosukKwon** [comment]
@zhuohan123 What do you think about adding this kind of non-OpenAI features to the OpenAI server?

**Comment by vdobrovolskii** [comment]
Can we please make it happen? Repetition penalty is a very useful parameter to have and supporting it in the OpenAI server makes life a lot easier

File: vllm/entrypoints/openai/protocol.py [hunk 2]
```diff
@@ -57,12 +57,14 @@ class ChatCompletionRequest(BaseModel):
     messages: Union[str, List[Dict[str, str]]]
     temperature: Optional[float] = 0.7
     top_p: Optional[float] = 1.0
+    min_p: Optional[float] = 0.0
```
Comment (zhuohan123) [inline vllm/entrypoints/openai/protocol.py:60]: Can you move the parameters to the section below under
```python
    # Additional parameters supported by vLLM
```

**Review by zhuohan123 — APPROVED** [review]
Thank you for your PR! Left a small comment. @WoosukKwon We should include this in the openai API server, since we already have some parameters that the original server do not have (under `# Additional parameters supported by vLLM`)

File: vllm/entrypoints/openai/protocol.py (hunk 2 above)
Comment (esmeetu) [reply vllm/entrypoints/openai/protocol.py]: Didn't notice that..:) Resolved.

**Pull request merged**